### PR TITLE
[WNMGDS-2095] Fix disabled checked choices

### PR DESCRIPTION
**Pull request by pwolfert**
## Summary

https://jira.cms.gov/browse/WNMGDS-2095

- Fix disabled checked choices not looking disabled
- Improve support for them in forced-colors mode
- Created two new stories for disabled choices so we can capture this scenario in visual regression tests

## How to test

1. Check the new stories in Storybook (and in forced-colors mode)
2. Look at the new screenshots

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title`

### If this is a change to code:

- [x] Created or updated unit tests to cover the logic added
- [x] If necessary, updated unit-test snapshots (`yarn test:unit:update`) and browser-test snapshots (`yarn test:browser:update`)

### CODE REVIEW

**Comment by pwolfert** [comment]
I might follow up on this with a pull request that separates the checkbox and radio styles more. Right now they're just really intertwined and complex

File: packages/design-system/src/styles/components/_Choice.scss [hunk 1]
```diff
@@ -165,6 +173,23 @@ label,
...
+      &::before {
+        box-shadow: inset 0 0 4em 1em $choice__border-color--disabled;
+      }
+
+      @media (-ms-high-contrast: active), (forced-colors: active) {
+        border-color: GrayText;
+
+        &::before {
+          background-color: GrayText;
+          box-shadow: inset 0 0 4em 1em GrayText;
```
Comment (zarahzachz) [inline packages/design-system/src/styles/components/_Choice.scss:189]: `box-shadow` isn't respected by HCM, so I don't think this is doing anything here.

**Review by zarahzachz — APPROVED** [review]
Looks good! Had one comment, but works pretty well overall.

**Review by forrestbaer — APPROVED** [review]
Looks good!

File: packages/design-system/src/styles/components/_Choice.scss (hunk 1 above)
Comment (pwolfert) [reply packages/design-system/src/styles/components/_Choice.scss]: Good call!

**Pull request merged**